src/normalizers/keywordProtector.js

```diff
@@ -0,0 +1,49 @@
+import { regexMatchingDecoderMaker } from './regex.js';
+
+export default class KeywordProtector {
+    constructor(keywordList, keywordMap) {
+        this.keywordList = keywordList;
+        this.keywordMap = keywordMap;
+        this.defaultEncoder = this.regexMatchingEncoderMaker(
+            'protectedStringsDecoder',
+            /(?<protector>protector:\w+)/g,
+            this.keywordList.reduce((out, str) => { out[`protector:${str}`] = str; return out}, {})
+        );
+    
+        this.mappedEncoder = this.regexMatchingEncoderMaker(
```

Comment:
can you put some comments to explain usage? what's the difference between a `defaultEncoder` and a `mappedEncoder`? when would i use one or the other? i wasn't aware we needed 2

Reply:
With the change to one input, we only need one encoder now.  I deleted the other one.